src/message/MentionedUserNotSubscribed.js

```diff
@@ -0,0 +1,71 @@
+/* @flow strict-local */
+
+import React, { Component } from 'react';
+import { View, TouchableOpacity } from 'react-native';
+
+import type { Auth, Stream, Dispatch, Narrow, UserOrBot, GetText } from '../types';
+import { withGetText } from '../boot/TranslationProvider';
+import { connect } from '../react-redux';
+import * as api from '../api';
+import { ZulipButton, RawLabel } from '../common';
+import { getAuth, getStreamInNarrow, getUserForId } from '../selectors';
+import styles from '../styles';
+
+type SelectorProps = {|
+  auth: Auth,
+  stream: { ...Stream },
+  user: UserOrBot,
+|};
+
+type Props = $ReadOnly<{|
+  narrow: Narrow,
+  userId: number,
+  onDismiss: (user: UserOrBot) => void,
+  dispatch: Dispatch,
+  ...SelectorProps,
+  _: GetText,
+|}>;
+
+class MentionedUserNotSubscribed extends Component<Props> {
+  subscribeToStream = () => {
+    const { auth, stream, user } = this.props;
+    api.subscriptionAdd(auth, [{ name: stream.name }], [user.email]);
+    this.handleDismiss();
+  };
+
+  handleDismiss = () => {
+    const { user, onDismiss } = this.props;
+    onDismiss(user);
+  };
+
+  render() {
+    const { user, _ } = this.props;
+
+    const alertTitle = _.intl.formatMessage(
+      {
+        id: '{username} will not be notified unless you subscribe them to this stream.',
+        defaultMessage: '{username} will not be notified unless you subscribe them to this stream.',
+      },
+      { username: user.full_name },
+    );
+
+    return (
+      <View>
+        <TouchableOpacity onPress={this.handleDismiss} style={styles.mentionedUserNotSubscribed}>
+          <RawLabel style={styles.mentionedUserNotSubscribedText} text={alertTitle} />
```

Comment:
Hmm, I see. Let's use `Label` instead -- that's how we normally handle translation, so that will be more explicit about how we're translating this string.

Note that its argument is a `LocalizableText` -- you can pass an object there like `{ text: '{username} will not …', values: { username: user.full_name } }`.

(That probably is not an awesome way to write that interface, and we should make it take a `values` prop instead. But that's an independent improvement -- the API should work just fine already.)

I think that will also be a bit simpler in this code than calling `_.intl.formatMessage` directly is.

Reply:
Changed to `Label`.